src/classes/PS_CommitmentRequest.cls

```diff
@@ -178,16 +190,22 @@ public inherited sharing class PS_CommitmentRequest {
     */
     @TestVisible
     private void setDonorInfo(RequestBody reqBody, npe03__Recurring_Donation__c rd) {
-        Boolean isContactDonor = rd.npe03__Contact__c != null
+        Contact contactDonor = rd.npe03__Contact__c != null
+            ? getContact(rd.npe03__Contact__c)
+            : null;
+
+        Boolean isContactDonor = contactDonor != null
```

Comment:
I find the `isContactDonor` and the `contactDonor` variables confusing. I think changing `isContactDonor` to `isHouseholdDonor` will be clearer.